stdlib/public/core/EmbeddedRuntime.swift

```diff
@@ -187,6 +187,20 @@ func isValidPointerForNativeRetain(object: Builtin.RawPointer) -> Bool {
 public func swift_setDeallocating(object: Builtin.RawPointer) {
 }
 
+@_cdecl("swift_dynamicCastClass")
+public func swift_dynamicCastClass(object: UnsafeRawPointer, targetMetadata: UnsafeRawPointer) -> UnsafeRawPointer? {
+  let sourceObj = object.assumingMemoryBound(to: HeapObject.self)
+  var type = sourceObj.pointee.metadata.assumingMemoryBound(to: ClassMetadata.self)
```

Comment:
I think this might need to be a call to `_swift_embedded_get_heap_object_metadata_pointer` instead, to respect ptrauth correctly?